### ISSUE TITLE
Why Batchnorm at the final network output?

### ISSUE THREAD

**Issue by amostayed** [yaoing/DAN#14]
Hi,
Thanks for this excellent repository. Very easy to follow. A question about the implementation though:
I don't often see a batchnorm layer before the softmax loss in classifier networks. Any specific reason you have it? What if you train without the last batchnorm? I had a quick check without the BN and the cross-entropy loss after the first training batch was : (on a very small private dataset with 4 classes)  ~313. When I did the same with the BN the value was ~1.94, which is more typical of cross entropy loss at the beginning of the training.

**Comment by yaoing** [comment]
Hi, thanks for your interest!
You are right, the BN put after FC layer is odd. The reason for using BN is that I found the model could'n learn very well when setting FC as the output layer. And I noticed there are several works add normalization at the end network as a trick. I just try it, and it work. This may be related to the structure of the entire AFN, and it's hard for me to explain.
